packages/canary-web-components/src/components/ic-tree-view/ic-tree-view.tsx

```diff
@@ -245,17 +255,55 @@ export class TreeView {
   private getAllTreeItems(element: HTMLElement): HTMLIcTreeItemElement[] {
     const treeItems: HTMLIcTreeItemElement[] = [];
 
-    const collectTreeItems = (el: HTMLElement) => {
+    const collectTreeItemsFromSlottedContent = (el: HTMLElement) => {
       Array.from(el.children).forEach((child) => {
         if (child.tagName === this.treeItemTag) {
           treeItems.push(child as HTMLIcTreeItemElement);
         }
 
-        collectTreeItems(child as HTMLElement);
+        collectTreeItemsFromSlottedContent(child as HTMLElement);
+      });
+    };
+
+    const collectTreeItemsFromData = (
+      items: IcTreeItemOptions[],
+      parentElement: HTMLElement
+    ) => {
+      Array.from(parentElement.children).forEach((child) => {
+        if (child.tagName === this.treeItemTag) {
+          parentElement.removeChild(child);
+        }
+      });
+
+      items.forEach((item) => {
+        const treeItem = document.createElement(
+          this.treeItemTag
+        ) as HTMLIcTreeItemElement;
+        const { children, icon, ...props } = item;
+        Object.assign(treeItem, props);
+
+        if (icon) {
+          const iconSlot = document.createElement("div");
+          iconSlot.setAttribute("slot", "icon");
+          iconSlot.innerHTML = icon;
+          treeItem.appendChild(iconSlot);
+        }
+
+        parentElement.appendChild(treeItem);
+        treeItems.push(treeItem);
+
+        if (item.children && item.children.length > 0) {
```

Comment:
```suggestion
        if (children && children.length > 0) {
```

Children is assigned above when item is destructured but doesn't appear to be used? Unless I'm mistaken.